eth/gasprice/gasprice.go

```diff
@@ -133,6 +141,11 @@ func NewOracle(backend OracleBackend, params Config) *Oracle {
 		defaultBasePrice = DefaultBasePrice
 		log.Warn("Sanitizing invalid gasprice oracle default base price", "provided", params.DefaultBasePrice, "updated", defaultBasePrice)
 	}
+	defaultGasTipCap := params.DefaultGasTipCap
+	if defaultGasTipCap == nil || defaultGasTipCap.Int64() < 0 {
```

Comment:
Just to make sure: we allow setting it to 0?

Reply:
similar to DefaultBasePrice (0 wei), it's valid for a txn with 0 gas tip, too. There are some gotchas to block these txns in tx pool's default setting (e.g. `DefaultTxPoolConfig.PriceLimit` for remote txns).

so for engineering tricks, it's also a good practice to make it non-zero. I'm OK with both. which do you prefer?

Reply:
after rethinking, it's more practical to enforce positive gas tip, thus changed by this commit: [5e3cbb3](https://github.com/scroll-tech/go-ethereum/pull/1096/commits/5e3cbb30f8e1ae9885b5df8be63807b2d85cefdb)